include/alpaka/queue/cuda-hip/QueueUniformCudaHipRt.hpp

```diff
@@ -71,8 +85,11 @@ namespace alpaka
                 // Make sure all pending async work is finished before destroying the stream to guarantee determinism.
                 // This would not be necessary for plain CUDA/HIP operations, but we can have host functions in the
                 // stream, which reference this queue instance and its CallbackThread. Make sure they are done.
-                ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK_NOEXCEPT(TApi::streamSynchronize(m_UniformCudaHipQueue));
-                ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK_NOEXCEPT(TApi::streamDestroy(m_UniformCudaHipQueue));
+                if(m_isOwning)
+                {
+                    ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK_NOEXCEPT(TApi::streamSynchronize(m_UniformCudaHipQueue));
+                    ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK_NOEXCEPT(TApi::streamDestroy(m_UniformCudaHipQueue));
+                }
```

Comment:
I agree that, if the queue is not owned, it should not be destroyed in the destructor.
However, maybe it should still be synchronised ?
@sbaldu I guess it mostly depends on how you plan to use the non-owning queues.

```suggestion
                ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK_NOEXCEPT(TApi::streamSynchronize(m_UniformCudaHipQueue));
                if(m_isOwning)
                {
                    ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK_NOEXCEPT(TApi::streamDestroy(m_UniformCudaHipQueue));
                }
```